src/components/LocationSelector/__tests__/LocationSelector.spec.tsx

```diff
@@ -0,0 +1,106 @@
+                lat: 52.09073739999999,
+            },
+        } as LocationSelectorLocation,
+    ];
+
+    const minRadius = 1;
+    const maxRadius = 100;
+    const radiusStep = 1;
+    const radiusDefaultValue = 2;
+    const renderRadiusLabel = (r) => `+ ${r} km`;
+    const country = 'NL';
+    const placeTypes = ['(regions)'];
+
+    const inputPlaceholder = 'inputPlaceholder';
+    const noSuggestionsPlaceholder = 'noSuggestionsPlaceholder';
+    const doneLabel = 'doneLabel';
+
+    const onAddLocationMock = jest.fn();
+    const onUpdateLocationMock = jest.fn();
+    const onRemoveLocationMock = jest.fn();
+    const onLocationAutocompleteErrorMock = jest.fn();
+
+    let view: RenderResult;
+
+    beforeEach(() => {
+        view = render(
+            <LocationSelector
+                apiKey="apiKey"
+                language="en"
+                radiusUnits="km"
+                radiusDefaultValue={radiusDefaultValue}
+                modalContentLabel="Location selection dialog"
+                clearTooltipLabel="Clear"
+                inputPlaceholder={inputPlaceholder}
+                hasRadius
+                minRadius={minRadius}
+                maxRadius={maxRadius}
+                radiusStep={radiusStep}
+                renderRadiusLabel={renderRadiusLabel}
+                onRemoveLocation={onRemoveLocationMock}
+                doneLabel={doneLabel}
+                country={country}
+                placeTypes={placeTypes}
+                noSuggestionsPlaceholder={noSuggestionsPlaceholder}
+                showCountryInSuggestions
+                onLocationAutocompleteError={onLocationAutocompleteErrorMock}
+                onUpdateLocation={onUpdateLocationMock}
+                selectedLocations={selectedLocations}
+                onAddLocation={onAddLocationMock}
+                onRemoveAllLocations={jest.fn()}
+            />
+        );
+    });
+
+    it('should render LocationSelector propertly when modal is closed', () => {
+        expect(view.container).toMatchSnapshot();
+        expect(screen.getByRole('img')).toBeInTheDocument();
+        expect(screen.getAllByRole('button')).toHaveLength(2);
+    });
+
+    it('should render LocationSelector properly when modal is open', async () => {
+        const user = userEvent.setup();
+        await user.click(screen.getByRole('button', { name: 'inputPlaceholder' }));
+
+        expect(view.container).toMatchSnapshot();
+        expect(screen.getByRole('img')).toBeInTheDocument();
+        expect(screen.getAllByRole('button')).toHaveLength(2);
+        expect(screen.getByRole('status')).toBeInTheDocument();
+    });
+
+    it('should open and close modal when requested', async () => {
+        const user = userEvent.setup();
+        const button = screen.getByRole('button', { name: 'inputPlaceholder' });
+        expect(button).not.toHaveAttribute('isOpen');
+        await user.click(button);
+
+        // Since JSDom cannot click outside of component,
```

Comment:
The code after the comment doesn't reflect the comment. This comment was migrated from the enzyme code but in there, there was a call to onRequestClose.

Reply:
:+1: